Update documentation for generate-features goal (#1522)
OpenLiberty/ci.maven: docs/generate-features.md

#### generate-features
---
Scan the class files of an application and create a new `generated-features.xml` Liberty configuration file containing the features the application requires.

> This feature is best accessed through [dev mode](dev.md). When you start `liberty:dev` your application will be compiled and the class files will be scanned to verify that all the required Liberty features are included in your server configuration. Then as you work, dev mode will continue to monitor the project to confirm the Liberty features configured are up to date. If you implement a new interface in Java, the scanner will determine if that API is connected to a Liberty feature, then update the server configuration and install the feature. If you remove a feature from `server.xml`, dev mode will determine if that feature is actually necessary, and if so, add it to the generated configuration file as described below.

Reviewer: ```suggestion
This feature is best accessed through [dev mode](dev.md) during development. When you start `liberty:dev` your application will be compiled and the class files will be scanned to verify that all the required Liberty features are included in your server configuration. Then as you work, dev mode will continue to monitor the project to confirm the Liberty features configured are up to date. If you implement a new interface in Java, the scanner will determine if that API is connected to a Liberty feature, then update the server configuration and install the feature. If you remove a feature from `server.xml`, dev mode will determine if that feature is actually necessary, and if so, add it to the generated configuration file as described below.
```

##### Lifecycle

This goal is not part of the Maven lifecycle, so to use it in your build you will need to understand its dependencies. Since it will scan the class files of your application, it must be run after the `compile` goal. The list of features that it generates will be used by the `liberty:create` and the `liberty:install-feature` goals, so run this goal first.

If this goal detects Liberty features used in your project but not present in your Liberty configuration, it will create a new file `configDropins/overrides/generated-features.xml` in the `src/main/liberty/config` directory of your project. The `generated-features.xml` file will contain a list of features required for your project. If the `generated-features.xml` file has been created in the past and no additional features have been detected, this file will be retained.

> This goal examines the `pom.xml` dependencies to determine which version of Java/JakartaEE EE and MicroProfile you may be using. Compatible features will then be generated. 

Reviewer: ```suggestion
This goal examines the `pom.xml` dependencies to determine which version of Jakarta EE, MicroProfile or Java EE API you may be using. Compatible features will then be generated. 
```

> For Java EE this goal looks for a `javax:javaee-api` dependency with versions `6.0`, `7.0` or `8.0`. For Jakarta EE this goal looks for a `jakarta.platform:jakarta.jakartaee-api` dependency with version `8.0`.

Reviewer: ```suggestion
For Jakarta EE API, this goal looks for a `jakarta.platform:jakarta.jakartaee-api` dependency with version `8.0`.
```

> For MicroProfile this goal looks for a `org.eclipse.microprofile:microprofile` dependency and generates features according to the version number.

Reviewer: ```suggestion
For MicroProfile API, this goal looks for a `org.eclipse.microprofile:microprofile` dependency and generates features according to the version number.

For Java EE API, this goal looks for a `javax:javaee-api` dependency with versions `6.0`, `7.0` or `8.0`. 
```

> For example, if you have the following Jakarta EE and MicroProfile dependencies in your `pom.xml` features compatible with Jakarta EE `8.0` and MicroProfile `4.1` will be generated.

Reviewer: ```suggestion
For example, if you have the following Jakarta EE and MicroProfile dependencies in your `pom.xml` file, features compatible with Jakarta EE `8.0` and MicroProfile `4.1` will be generated.
```

##### Lifecycle

This goal is not part of the Maven lifecycle, so to use it in your build you will need to understand its dependencies. Since it will scan the class files of your application, it must be run after the `compile` goal. The list of features that it generates will be used by the `liberty:create` and the `liberty:install-feature` goals, so run this goal first.

If this goal detects Liberty features used in your project but not present in your Liberty configuration, it will create a new file `configDropins/overrides/generated-features.xml` in the `src/main/liberty/config` directory of your project. The `generated-features.xml` file will contain a list of features required for your project. If the `generated-features.xml` file has been created in the past and no additional features have been detected, this file will be retained.

> If you are using [devc](dev.md#devc-container-mode), ensure that the `generated-features.xml` configuration file is copied via your Dockerfile.

Reviewer 2: ```suggestion
If you are using [devc](dev.md#devc-container-mode), ensure that the `generated-features.xml` configuration file is copied to your Docker image via your Dockerfile.
```

```dockerfile
COPY --chown=1001:0  target/liberty/wlp/usr/servers/defaultServer/configDropins/overrides/generated-features.xml /config/configDropins/overrides/
```

This goal examines the `pom.xml` dependencies to determine which version of Jakarta EE, MicroProfile or Java EE API you may be using. Compatible features will then be generated. 

> For Jakarta EE API, this goal looks for a `jakarta.platform:jakarta.jakartaee-api` dependency with version `8.0`.

Reviewer 2: ```suggestion
For Jakarta EE API, this goal looks for a `jakarta.platform:jakarta.jakartaee-api` dependency with version `8.0.0`.
```

For MicroProfile API, this goal looks for a `org.eclipse.microprofile:microprofile` dependency and generates features according to the version number.

For Java EE API, this goal looks for a `javax:javaee-api` dependency with versions `6.0`, `7.0` or `8.0`. 

> For example, if you have the following Jakarta EE and MicroProfile dependencies in your `pom.xml` file, features compatible with Jakarta EE `8.0` and MicroProfile `4.1` will be generated.

Reviewer 2: ```suggestion
For example, if you have the following Jakarta EE and MicroProfile dependencies in your `pom.xml` file, features compatible with Jakarta EE `8.0.0` and MicroProfile `4.1` will be generated.
```

If this goal detects Liberty features used in your project but not present in your Liberty configuration, it will create a new file `configDropins/overrides/generated-features.xml` in the `src/main/liberty/config` directory of your project. The `generated-features.xml` file will contain a list of features required for your project. If the `generated-features.xml` file has been created in the past and no additional features have been detected, this file will be retained.

If you are using [devc](dev.md#devc-container-mode), ensure that the `generated-features.xml` configuration file is copied via your Dockerfile.
```dockerfile

> COPY --chown=1001:0  target/liberty/wlp/usr/servers/defaultServer/configDropins/overrides/generated-features.xml /config/configDropins/overrides/

Reviewer 3: Again the special suggestion for Linux.
